lib/spack/spack/compilers/__init__.py

```diff
@@ -125,6 +125,105 @@ def get_compiler_config(scope=None, init_config=True):
     return config
 
 
+def get_compiler_config_from_packages(scope=None):
+    """Return the compiler configuration from packages.yaml"""
+    packages = []
+    config = spack.config.get("packages", scope=scope)
+
+    if not config:
+        return packages
```

Comment:
(minor) This is slightly clearer on returning an empty list:
```suggestion
    config = spack.config.get("packages", scope=scope)
    if not config:
        return []
        
    packages = []
```